zerver/actions/user_settings.py

```diff
@@ -447,3 +451,25 @@ def do_change_user_setting(
         # not deleted every previously synced draft - to do that use the DELETE
         # endpoint.
         Draft.objects.filter(user_profile=user_profile).delete()
+
+    # if user changes "Display my availability to other users when online" setting,
+    # then quickly show/hide them to other uses via do_update_user_presence()
+    # which calls send_event. see https://github.com/zulip/zulip/issues/21108
```

Comment:
This issue number is wrong/unrelated. I think I'll remove that in favor of explaining the issue anyway.